File: testdata/fixtures/php/CWE-327.php
<?php
/* CWE-327: Use of a Broken or Risky Cryptographic Algorithm.
 *
 * MD5 and SHA-1 have practical collisions, crypt() without an explicit
 * modern salt falls back to DES, and str_rot13 is not encryption at all.
 * Which of these is fatal depends on what the value is for, and nothing in
 * the call says - so the call is what gets reported.
 */

function digest($data, $file, $password)
{
    /* ruleid: weak-crypto */
    $a = md5($data);
    /* ruleid: weak-crypto */
    $b = sha1($data);
    /* ruleid: weak-crypto */
    $c = md5_file($file);
    /* ruleid: weak-crypto */
    $d = sha1_file($file);
    /* ruleid: weak-crypto */
    $e = crypt($password);
    /* ruleid: weak-crypto */
    $f = str_rot13($data);

    /* ok: weak-crypto */
    $g = hash('sha256', $data);
    /* ok: weak-crypto */
    $h = password_hash($password, PASSWORD_DEFAULT);
    /* ok: weak-crypto */
    $i = hash_hmac('sha256', $data, $key);

    return [$a, $b, $c, $d, $e, $f, $g, $h, $i];
}

File: testdata/fixtures/ocaml/CWE-327.ml
(* CWE-327: Use of a Broken or Risky Cryptographic Algorithm
 * Digest.* in OCaml is MD5 and must not be used for security.
 * The rule flags any use of Digest.bytes / Digest.string etc.
 *)

let vuln_digest () =
  (* ruleid: ocamllint-digest *)
  let _ = Digest.string "hello" in
  (* ruleid: ocamllint-digest *)
  let _ = Digest.file "data.txt" in
  (* ruleid: ocamllint-digest *)
  let _ = Digest.bytes (Bytes.of_string "abc") in
  ()

let safe_digest () =
  (* ok: ocamllint-digest *)
  let _ = "hello" in
  (* ok: ocamllint-digest *)
  let _ = Printf.printf "safe\n" in
  ()
